tests/post/steps/test_example_solution.py

```diff
@@ -0,0 +1,398 @@
+def is_imported_solution_archive(host, k8s_client, name):
+    # Solution is imported correctly if
+    # 1. Solution Configuration is created at /etc/metalk8s/solution.yaml
+    # 2. Solution Configuration has new archive path
+    # 3. Solutions ConfigMap has new Solution imported
+
+    with host.sudo():
+        if not host.file(SOLUTION_CONFIGURATION_FILE).exists:
+            assert False, 'No Solution Configuration file exist at {}'.format(
+                        SOLUTION_CONFIGURATION_FILE
+                    )
+    with host.sudo():
+        file_contents = host.file(SOLUTION_CONFIGURATION_FILE).content_string
+        if name not in file_contents:
+            assert False, 'No Solution ISO found in Solution Config {}'.format(
+                        SOLUTION_CONFIGURATION_FILE, name
+                    )
+    configmap = get_configmap(
+        k8s_client, SOLUTION_CONFIGMAP, SOLUTION_NAMESPACE
+    )
+    assert name in configmap.get('data', {}), \
+        '{} Configmap does not contain {}'.format(SOLUTION_CONFIGMAP, name)
+
+
+@then(parsers.parse("Solution '{name}' is activated"))
+def is_activated_solution(host, request, name):
+    with host.sudo():
+        output = host.check_output(
+            'salt-call --out=json pillar.get "{}"'.format(SOLUTION_PILLAR_KEY)
+        )
+    data = json.loads(output)['local']
+
+    assert name in data
+
+
+@then(parsers.parse("solution environment '{name}' is available"))
+def read_solution_environment(k8s_client, name):
+    solution_environment_obj = get_environment(k8s_client, name)
+    assert solution_environment_obj is not None
+
+
+@then(parsers.parse(
+    "we have '{value}' available '{path}' for Solution '{name}' Deployment in "
+    "environment '{environment}'"))
+def get_deployments(
+    context, k8s_appsv1_client, value, path, name, environment
+):
+    assert context.get('deployment_obj') is not None
+    assert literal_eval(value) == context.get('deployment_obj'), (
+        "Expected {} replicas for Deployment {}, got {}".format(
+            value, name, context.get('deployment_obj')
+        )
+    )
+
+
+@then(parsers.parse("we have no Solution '{name}' archive mounted"))
+def no_solution_mountpoint(host, name, k8s_client):
+    with host.sudo():
+        if name in host.mount_point.get_mountpoints():
+            assert False, 'Found Solution mountpoint {}'.format(name)
+
+
+@then(parsers.parse("we have no Solution environment '{name}'"))
+def no_solution_environment(k8s_client, name):
+    solution_environment_obj = get_environment(k8s_client, name)
+    assert solution_environment_obj is None
+
+
+@then(parsers.parse("we have no activated Solution '{name}'"))
+def no_activated_solution(host, request, name):
+    with host.sudo():
+        output = host.check_output(
+            'salt-call --out=json pillar.get "{}"'.format(SOLUTION_PILLAR_KEY)
+        )
+    data = json.loads(output)['local']
+
+    assert name not in data
```

Comment:
This function is a mostly a duplicate of `is_activated_solution()`, you should create an helper and use it in both functions.
And btw, you're not testing if the solution is activated, but if it's available.